megatron/data/gpt_dataset.py

```diff
@@ -354,13 +540,15 @@ def _build_index_mappings(name, data_prefix, documents, sizes,
     desc += f"Sequence length {seq_length}\n"
     desc += f"Random seed {seed}\n"
     desc += f"Split {splits_string}\n"
-    desc_hash = hashlib.md5(desc.encode('utf-8')).hexdigest()
+    desc += f"Shuffle original documents {shuffle_orig_docs}\n"
```

Comment:
change: add whether the documents were shuffled at the top level to the description hash. When searching for files, whether the top level assignments were shuffled will also be matched when megatron is deciding whether to construct the dataset or load in existing indices